x-pack/plugins/observability/server/lib/rules/custom_threshold/lib/metric_query.ts

```diff
@@ -19,16 +22,23 @@ import { createBucketSelector } from './create_bucket_selector';
 import { wrapInCurrentPeriod } from './wrap_in_period';
 import { getParsedFilterQuery } from '../../../../utils/get_parsed_filtered_query';
 
-export const calculateCurrentTimeframe = (
+export const calculateCurrentTimeFrame = (
   metricParams: CustomMetricExpressionParams,
   timeframe: { start: number; end: number }
-) => ({
-  ...timeframe,
-  start: moment(timeframe.end).subtract(metricParams.timeSize, metricParams.timeUnit).valueOf(),
-});
+) => {
+  const isRateAgg = metricParams.metrics.some((metric) => metric.aggType === Aggregators.RATE);
```

Comment:
How would it work if we have multiple aggregations and only one of them is rate? Will it use the double bucket size for all aggregations?

Reply:
@maryam-saeidi I was investigating the 2-second question, which is not easy. The outcome is even the Metric rule could have the same behavior, i.e. 2-3 seconds difference in the first time range filter 
e.g.:
{
   "index":"metrics-*,metricbeat-*,high*",
   "allow_no_indices":true,
   "ignore_unavailable":true,
   "body":{
      "track_total_hits":true,
      "query":{
         "bool":{
            "filter":[
               {
                  "range":{
                     "@timestamp":{
                        "gte":"2024-01-17T16:28:45.298Z",
                        "lte":"2024-01-17T16:30:48.302Z"
                     }
                  }
               },
               {
                  "exists":{
                     "field":"host.network.egress.bytes"
                  }
               }
            ]
         }
      },
      "size":0,
      "aggs":{
         "all":{
            "filters":{
               "filters":{
                  "all":{
                     "match_all":{
                        
                     }
                  }
               }
            },
            "aggs":{
               "currentPeriod":{
                  "filters":{
                     "filters":{
                        "all":{
                           "range":{
                              "@timestamp":{
                                 "gte":"2024-01-17T16:28:48.302Z",
                                 "lte":"2024-01-17T16:30:48.302Z"
                              }
                           }
                        }
                     }
                  },
                  "aggs":{
                     "aggregatedValue_first_bucket":{
                        "filter":{
                           "range":{
                              "@timestamp":{
                                 "gte":"2024-01-17T16:28:48.302Z",
                                 "lt":"2024-01-17T16:29:48.302Z"
                              }
                           }
                        },
                        "aggs":{
                           "maxValue":{
                              "max":{
                                 "field":"host.network.egress.bytes"
                              }
                           }
                        }
                     },
                     "aggregatedValue_second_bucket":{
                        "filter":{
                           "range":{
                              "@timestamp":{
                                 "gte":"2024-01-17T16:29:48.302Z",
                                 "lt":"2024-01-17T16:30:48.302Z"
                              }
                           }
                        },
                        "aggs":{
                           "maxValue":{
                              "max":{
                                 "field":"host.network.egress.bytes"
                              }
                           }
                        }
                     }
                  }
               },
               "aggregatedValue":{
                  "bucket_script":{
                     "buckets_path":{
                        "first":"currentPeriod['all']>aggregatedValue_first_bucket.maxValue",
                        "second":"currentPeriod['all']>aggregatedValue_second_bucket.maxValue"
                     },
                     "script":"params.second > 0.0 && params.first > 0.0 && params.second > params.first ? (params.second - params.first) / 60: null"
                  }
               },
               "shouldWarn":{
                  "bucket_script":{
                     "buckets_path":{
                        
                     },
                     "script":"0"
                  }
               },
               "shouldTrigger":{
                  "bucket_script":{
                     "buckets_path":{
                        "value":"aggregatedValue"
                     },
                     "script":{
                        "source":"params.value > params.threshold ? 1 : 0",
                        "params":{
                           "threshold":0.01
                        }
                     }
                  }
               }
            }
         }
      }
   }
}

Reply:
But the other time range filters are exactly 2 seconds like the Custom Threshold Rule.

Reply:
Difference in seconds is related to `lastPeriod` ([more info](https://elastic.slack.com/archives/C05UT5PP1EF/p1706697683095239))